src/matching/match_basins.jl

```diff
@@ -0,0 +1,81 @@
+# TODO: I don't know how this fits into the new interface...
+
+"""
+    MatchByBasinOverlap(threshold = Inf)
+
+A special matcher that matches IDs given full basins of attraction.
+It matches IDs of attractors whose basins of attraction before and after `b₋, b₊`
+have the most overlap (in pixels). This overlap is normalized in 0-1 (with 1 meaning
+100% overlap of pixels). The `threshold` in this case is compared to the inverse
+of the overlap, and basins whose overlap is less than `1/threshold` are guaranteed
+to get assined different IDs.
+For example: for `threshold = 2` basins that have less than 50% overlap get
+different IDs guaranteed.
+
+The input for this matcher in [`replacement_map`](@ref)
+should be dictionaries mapping IDs to vectors of cartesian indices,
+where the indices mean which parts of the state space belong to which ID
+
+This information can be derived from the basins of attraction (`Array`),
+i.e., the direct output of [`basins_of_attraction`](@ref).
+For convenience, as well as backwards compatibility, when using
+[`replacement_map`](@ref) with this mapper you may provide two `Array`s `b₊, b₋`
+representing basins of attraction after and before, and the conversion to dictionaries
+will happen internally as it is supposed to.
+To replace the `IDs` in `b₊` given the replacement map just call `replace!(b₊, rmap...)`,
+or use the in-place version [`replacement_map!`](@ref) directly.
+"""
+struct MatchByBasinOverlap
+    threshold::Float64 = Inf
+end
+
+function replacement_map(b₊::AbstractArray, b₋::AbstractArray, matcher::MatchByBasinOverlap; i = nothing)
```

Comment:
```suggestion
function replacement_map(b₊::AbstractArray, b₋::AbstractArray, matcher = MatchByBasinOverlap(); i = nothing)
```
Maybe leaving a default structure makes the function more user friendly.